xonsh/platform.py

```diff
@@ -0,0 +1,79 @@
+""" Module for platform-specific constants and implementations """
+
+import os
+import platform
+import sys
+
+
+#
+# Python
+#
+
+PYTHON_VERSION = sys.version_info[:3]
+
+if PYTHON_VERSION < (3, 5, 0):
+    def scandir(path):
+        """ Compatibility wrapper for os.scandir from Python 3.5+ """
+        return iter(os.listdir(path))
+else:
+    from os import scandir
+
+
+ON_ANACONDA = any(s in sys.version for s in ('Anaconda', 'Continuum'))
+
+
+#
+# OS
+#
+
+ON_DARWIN = ON_LINUX = ON_WINDOWS = None
+for name in ('Darwin', 'Linux', 'Windows'):
+    globals()['ON_' + name.upper()] = platform.system() == name
```

Comment:
i mean, what you refer to as 'normal' assignment is one code line more than this. it's three lines with a repeated pattern. which isn't reasonable with a higher amount of items.
discussion postponed until Commodore 64s are supported.